packages/amplify-category-notifications/src/channel-in-app-msg.ts

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable spaced-comment */
+/* eslint-disable no-param-reassign */
+/* eslint-disable prefer-arrow/prefer-arrow-functions */
+/* eslint-disable func-style */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import {
+  $TSAny, $TSContext, AmplifyCategories, amplifyFaultWithTroubleshootingLink, AmplifySupportedService,
+  IPluginCapabilityAPIResponse,
+  NotificationChannels,
+  stateManager,
+} from 'amplify-cli-core';
+
+/* eslint-disable @typescript-eslint/no-var-requires */
+import * as inquirer from 'inquirer';
+import ora from 'ora';
+import { printer } from 'amplify-prompts';
+import { Notifications } from './notifications-api';
+import {
+  invokeAnalyticsResourceToggleNotificationChannel,
+} from './plugin-client-api-analytics';
+import { IChannelAPIResponse, ChannelAction, ChannelConfigDeploymentType } from './channel-types';
+import { ChannelCfg } from './notifications-backend-cfg-channel-api';
+import {
+  buildPinpointChannelResponseError,
+  buildPinpointChannelResponseSuccess,
+  getPinpointAppStatusFromMeta, IPinpointAppStatus, IPinpointDeploymentStatus,
+} from './pinpoint-helper';
+
+const channelName = 'InAppMessaging';
+const channelViewName = ChannelCfg.getChannelViewName(channelName);
+const spinner = ora('');
+const deploymentType = ChannelConfigDeploymentType.DEFERRED;
+
+const NOOP_CFG_RESPONSE: IChannelAPIResponse = {
+  action: ChannelAction.CONFIGURE, //since config has not transformed into enable/disable, user has decided not to take any action.
+  channel: channelName,
+  response: {
+    pluginName: AmplifyCategories.NOTIFICATIONS,
+    resourceProviderServiceName: AmplifySupportedService.PINPOINT,
+    status: true,
+    capability: AmplifyCategories.NOTIFICATIONS,
+    subCapability: ChannelCfg.ChannelType.InAppMessaging,
+  },
+  deploymentType: ChannelConfigDeploymentType.DEFERRED,
+};
+
+/**
+ * Configure Pinpoint with configs and IAM roles
+ * @param {*} context amplify cli context
+ */
+export const configure = async (context: $TSContext) : Promise<IChannelAPIResponse> => {
+  if (await Notifications.ChannelCfg.isChannelEnabledNotificationsBackendConfig(channelName)) {
+    printer.info(`The ${channelViewName} channel is currently enabled`);
+    const answer = await inquirer.prompt({
```

Comment:
I'm not sure if this file is from a renamed js file but we should really try to avoid adding new code that uses inquirer. If it's not too much of a lift to migrate this to amplify prompts that would be highly preferred